### PR TITLE
Add sync mode to AgentWriter

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
## Description

Add a mode to the agent writer so that traces can be flushed out synchronously.

Pulls the sync mode changes out of #2092.

## Checklist
- [x] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] ~Library documentation is updated.~
- [x] ~[Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).~

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]

File: tests/integration/test_integration.py [hunk 1]
```diff
@@ -393,114 +386,11 @@ def test_span_tags():
     log.error.assert_not_called()
 
 
-@pytest.mark.skipif(AGENT_VERSION != "testagent", reason="Tests only compatible with a testagent")
```
Comment (Kyle-Verhoog) [inline tests/integration/test_integration.py:396]: This change was pulled into #2209

**Comment by Kyle-Verhoog** [comment]
Some of the snapshot tests will not pass until #2208 is merged

**Review by P403n1x87 — DISMISSED** [review]

File: tests/integration/test_integration_snapshots.py [hunk 2]
```diff
@@ -68,53 +88,88 @@ def process_trace(self, trace):
...
 
 
+@pytest.mark.parametrize(
+    "writer",
+    ("default", "sync"),
+)
 @snapshot(include_tracer=True)
-def test_sampling(tracer):
+def test_sampling(writer, tracer):
+    if writer == "sync":
```
Comment (majorgreys) [inline tests/integration/test_integration_snapshots.py:97]: nit we could parameterize the fixture: https://docs.pytest.org/en/stable/fixture.html#fixture-parametrize

**Review by majorgreys — DISMISSED** [review]

**Review by Yun-Kim — APPROVED** [review]

File: tests/integration/test_integration_snapshots.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply tests/integration/test_integration_snapshots.py:95]: ah true

**Pull request merged**